snuba/consumers/consumer_builder.py

```diff
@@ -121,11 +113,23 @@ def __init__(
                 )
             else:
                 self.replacements_topic = None
+                self.replacements_producer = None
+
+        if self.replacements_topic is not None:
+            self.replacements_producer = Producer(
+                build_kafka_producer_configuration(
+                    SnubaTopic(self.replacements_topic.name),
+                    bootstrap_servers=kafka_params.bootstrap_servers,
+                    override_params={
+                        "partitioner": "consistent",
+                        "message.max.bytes": 50000000,  # 50MB, default is 1MB)
+                    },
+                )
+            )
```

Comment:
nit: I think it's a bit clearer if you define `self.replacements_producer = None` in an else block starting on line 129. So that it is being defined in the same place in both cases. Same for the commit log producer.